pkg/operator/ceph/cluster/osd/envs.go

```diff
@@ -213,6 +213,19 @@ func osdActivateEnvVar() []v1.EnvVar {
 	return append(cephVolumeEnvVar(), monEnvVars...)
 }
 
+func getEnvFromSources() []v1.EnvFromSource {
+	optionalConfigMapRef := true
+
+	return []v1.EnvFromSource{
+		{
+			ConfigMapRef: &v1.ConfigMapEnvSource{
+				LocalObjectReference: v1.LocalObjectReference{Name: "rook-ceph-daemon-env-override"},
```

Comment:
Since this feature is currently only for OSDs, I'm thinking we should use an osd-specific name for the configmap. If this feature is added to other daemons in the future it seems likely we will want separate env for each daemon type
```suggestion
				LocalObjectReference: v1.LocalObjectReference{Name: "rook-ceph-osd-env-override"},
```

Reply:
Agreed, switched to `rook-ceph-osd-env-override`.